src/api.c

```diff
@@ -541,6 +545,25 @@ void nmd_release_frame(struct nmd_frame *frame)
     }
 }
 
+int nmd_release_frame_mediacodec(struct nmd_frame **framep)
```

Comment:
It's weird for a render function to be named only "release"; I would name it `render_and_release` or `render_release` to be honest, even if the if the underlying function is actually named that way. If it's too long, `mediacodec` could be shorten into `mc`.

One important thing here is that this function does not have a "render" argument, which mean the render operation is completely hidden.

I would also argue that in the function `AMediaCodec_releaseOutputBuffer` the "output" might actually be a verb (just like "release") and not an adjective applied to "buffer". (Even if that's not the case, the previous point still stands though).

Reply:
> It's weird for a render function to be named only "release"; I would name it `render_and_release` or `render_release` to be honest, even if the underlying AMediaCodec function is actually named that way. If it's too long, `mediacodec` could be shorten into `mc`.

Renamed to `render_and_release`. Maybe not the final name though.

> 
> One important thing here is that this function does not have a "render" argument, which mean the render operation is completely hidden.
> 
> I would also argue that in the function `AMediaCodec_releaseOutputBuffer` the "output" might actually be a verb (just like "release") and not an adjective applied to "buffer". (Even if that's not the case, the previous point still stands though).

In `releaseOutputBuffer`, "output" is not a verb, it's an adjective that applies to the kind of buffer you are releasing and maps directly to the MediaCodec API, see: https://developer.android.com/reference/android/media/MediaCodec#releaseOutputBuffer(int,%20boolean). While `release` is not clear about the underlying mechanism, ie: release the buffer to the codec and render it to the output surface. The documentation is pretty clear about it and this naming is familiar to anyone working with Mediacodec.

Reply:
I also wonder if we should expose the render parameter to make it consistent with the mediacodec api. The user can potentially discard the buffer and do not want it to be rendered on screen (it still has the option to call `nmd_release_frame` though)

Reply:
Since you say it's possible to just call `nmd_release_frame` maybe not worth it. I'm not a fan of bool as argument like this overall, I think it's bad design. A bunch of flags ok, but a bool… meh.

`render_release` is also fine with me if you want to drop the `and`.